src/vcHotkey.cpp

```diff
@@ -213,7 +213,7 @@ namespace vcHotkey
     {
       errors |= (1 << vcKES_Select);
 
-      if (pProgramState->currentKey)
+      if (pProgramState->currentKey && pProgramState->currentKey != vcMOD_Alt + SDL_SCANCODE_F4)
```

Comment:
`(vcMOD_Alt + SDL_SCANCODE_F4)`.

Reply:
> I'm not sure if we should be blocking specific combinations, or just not saving immediately upon user input instead. Saving when the user clicks "Close" would probably be better, this would also give them the option to click "Cancel" and have the changes revert.

The saving is triggered by quitting the application, which I don't think should change

Reply:
Will do, but I think we should probably block specific combinations as well, binding alt-f4 is probably never useful/intended

Reply:
> binding alt-f4 is probably never useful/intended

Alternatively, let the user decide what is useful/intended - from memory Alt+F4 is fine on macOS.

If you stop writing the key changes directly into the settings struct and instead have a struct for the "new bindings", when the user clicks the Close button you can then stamp the new bindings struct over the top of the settings struct. When the user uses Alt+F4 on Windows, it might momentarily display this change but the Close button isn't triggered. There's no need to block the user from setting anything if they're unable to click the Close button.